services/worker/src/worker/job_runners/parquet_and_dataset_info.py

```diff
@@ -528,10 +528,28 @@ def raise_if_too_big_from_external_data_files(
         base_path=builder.base_path, download_config=DownloadConfig(use_auth_token=hf_token)
     )
     try:
-        builder._split_generators(mock_dl_manager)
-    except NotImplementedError as err:
-        if "is not implemented in streaming mode." not in str(err):
-            raise
+        builder._split_generators(mock_dl_manager)  # type: ignore
+    except (requests.exceptions.RequestException, NotImplementedError) as error:
+        if isinstance(error, NotImplementedError):
+            # we can ignore the errors from functions not implemented in streaming mode like `.extract()` on TAR files
+            if "is not implemented in streaming mode." not in str(error):
+                raise
+        elif isinstance(error, requests.exceptions.HTTPError):
+            raise ExternalFilesSizeRequestHTTPError(
+                "Couldn't get the list of external files in `_split_generators` because a request failed"
```

Comment:
In the same vein, maybe give a higher level message that a user of the Hub can make sense of.

Also: pass the error to the constructor as `cause`, to get the traceback into the database.